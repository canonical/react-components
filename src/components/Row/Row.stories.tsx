import { Meta, StoryObj } from "@storybook/react";

import Row from "./Row";

const meta: Meta<typeof Row> = {
  component: Row,
  tags: ["autodocs"],

  argTypes: {
    children: {
      control: {
        type: "text",
      },
    },
  },
};

export default meta;

type Story = StoryObj<typeof Row>;

/**
 * See [Col](?path=/docs/components-col--docs) for grid details.
 */
export const Default: Story = {
  name: "Default",

  args: {
    children: "children...",
  },
};
