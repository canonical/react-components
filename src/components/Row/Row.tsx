import classNames from "classnames";
import React from "react";
import type { HTMLProps, ReactNode } from "react";
import type { ClassName, PropsWithSpread } from "types";

export type Props = PropsWithSpread<
  {
    /**
     * The content of the row.
     */
    children: ReactNode;
    /**
     * Optional class(es) to pass to the wrapping div element.
     */
    className?: ClassName;
  },
  HTMLProps<HTMLDivElement>
>;

/**
 * This is a [React](https://reactjs.org/) component for use within the Vanilla [Grid](https://docs.vanillaframework.io/patterns/grid/).
 *
 * Vanilla has a responsive grid using a combination of rows and columns.
 */
const Row = ({ children, className, ...props }: Props): JSX.Element => (
  <div className={classNames(className, "row")} {...props}>
    {children}
  </div>
);

export default Row;
