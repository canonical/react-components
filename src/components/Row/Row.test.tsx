import { render, screen } from "@testing-library/react";
import React from "react";

import Row from "./Row";

describe("Row ", () => {
  it("renders", () => {
    render(<Row data-testid="row">Test content</Row>);
    expect(screen.getByTestId("row")).toMatchSnapshot();
  });

  it("can add additional classes", () => {
    render(
      <Row data-testid="row" className="extra-class">
        Test content
      </Row>
    );
    const row = screen.getByTestId("row");
    expect(row).toHaveClass("row");
    expect(row).toHaveClass("extra-class");
  });
});
