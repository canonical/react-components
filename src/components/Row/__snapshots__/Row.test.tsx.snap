// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row  renders 1`] = `
<div
  className="row"
>
  Test content
</div>
`;
